src/moin/storage/middleware/indexing.py

```diff
@@ -1240,7 +1254,8 @@ def store_revision(self, meta, data, overwrite=False,
         data.seek(0)  # rewind file
         backend_name, revid = backend.store(meta, data)
         meta[REVID] = revid
-        self.indexer.index_revision(meta, content, backend_name)
+        self.indexer.index_revision(meta, content, backend_name, force_latest=not overwrite)
+        gc.collect()  # triggers close of index files from is_latest search
```

Comment:
let me know what you think of adding `gc.collect()` here, it solves the test failure seen in https://github.com/bylsmad/moin/actions/runs/5102359184/jobs/9171871358 

it is also possible to solve the test failure by putting the `gc.collect` just before `destroy_app() ` in src.moin.conftest
but I was worried this would mask potential issues where unclosed files could lead to "too many open files" exception

Reply:
I will defer to you as where to put gc.collect(). 

The code seems to work well, but tests on Windows 10 came up with one error:

    ================================== FAILURES ===================================
    ________________________ TestSiteCrawl.test_home_page _________________________

    self = <moin.cli._tests.test_scrapy_crawl.TestSiteCrawl object at 0x000001F70A1768F0>
    crawl_results = []

        def test_home_page(self, crawl_results):
    >       assert len(crawl_results) > 0
    E       assert 0 > 0
    E        +  where 0 = len([])

    C:\git-david\moin\src\moin\cli\_tests\test_scrapy_crawl.py:54: AssertionError
    ----------------------------- Captured log setup ------------------------------

Reply:
Could you attach these files from the failed run:

_test_artifacts/crawl.log
_test_artifacts/server-crawl.log
_test_artifacts/crawl.csv